buildlib/rdma_functions.cmake

```diff
@@ -269,9 +269,16 @@ endfunction()
 # Generate a pkg-config file
 function(rdma_pkg_config PC_LIB_NAME PC_LIB_PRIVATE PC_INCLUDE_PRIVATE PC_REQUIRES)
   set(PC_LIB_NAME "${PC_LIB_NAME}")
+  # CMake separates ldflags with a semicolon, cannot be used in a command
+  string(REPLACE ";" " " PC_LIB_PRIVATE "${PC_LIB_PRIVATE}")
   set(PC_LIB_PRIVATE "${PC_LIB_PRIVATE}")
   set(PC_INCLUDE_PRIVATE "${PC_INCLUDE_PRIVATE}")
   set(PC_REQUIRES "${PC_REQUIRES}")
   configure_file(${CMAKE_SOURCE_DIR}/buildlib/template.pc.in ${CMAKE_CURRENT_BINARY_DIR}/lib${PC_LIB_NAME}.pc @ONLY)
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/lib${PC_LIB_NAME}.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+  # With IN_PLACE=1 the install step is not ran, so copy the files manually
+  if (IN_PLACE)
+    configure_file(${CMAKE_CURRENT_BINARY_DIR}/lib${PC_LIB_NAME}.pc ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/pkgconfig/lib${PC_LIB_NAME}.pc COPYONLY)
+  endif()
```

Comment:
There is no real need to have both .pc files.. Just always emit them to build/lib/pkgconfig

If in_place is set then they are valid for use without installation, otherwise they are the to-be-installed versions..

Reply:
Also the IN_PLACE .pc files should have -Wl,-rpath, I think ?

Reply:
> There is no real need to have both .pc files.. Just always emit them to build/lib/pkgconfig
>
> If in_place is set then they are valid for use without installation, otherwise they are the to-be-installed versions..

Ok, done.

> Also the IN_PLACE .pc files should have -Wl,-rpath, I think ?

Good point, yes it should:

```
prefix=/home/bluca/git/rdma-core/build
exec_prefix=${prefix}
libdir=${prefix}/lib
includedir=${prefix}/include

Name: librdmacm
Description: RDMA Core Userspace Library
URL: https://github.com/linux-rdma
Version: 1.1.20.0
Libs: -L${libdir} -lrdmacm -Wl,-rpath,/home/bluca/git/rdma-core/build/lib
Libs.private: 
Cflags: -I${includedir} 
Requires:
```
```
$ gcc /tmp/test.c `PKG_CONFIG_PATH=$PWD/lib/pkgconfig pkg-config --libs --cflags libibverbs`
$ objdump -x a.out | grep RUN
  RUNPATH              /home/bluca/git/rdma-core/build/lib
```